### PR TITLE
Fix 67 reorganize prompt

### LINKED ISSUE #67: Reorganize Prompt
Look at the following pseudocode:

```
run() {
    var config = prompt();
    initProject(config);
}
```

- [ ] All prompts should be called from that `prompt()` method.
- [ ] Heroku, GitHub, and TravisCI, Docker should not be required. 
- [ ] Reorganize the prompt so that if they don't have GitHub, they are not prompted for TravisCI
- [ ] Add a check for the githubUsername and githubPassword before calling setupGitHub()

### PR DESCRIPTION
**Pull request by jackcmeyer**
closes #67 
* reorganizes the prompt for a better user workflow
* added clients for docker and heroku so we can validate credentials
* refactored a lot of error messaging to stop using constants
* separated static file creation and third party account access/creation
* added some user reporting

### CODE REVIEW

File: src/clients/heroku.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const herokuApiAccept = ' application/vnd.heroku+json; version=3';
+
+/**
+ * Returns the string used for the basic authorization header in a GET request.
+ *
+ * @param username the username
+ * @param password the password
+ * @returns {string} the basic authentication string
+ */
+export function basicAuthorization(username, password) {
```
Comment (enclark) [inline src/clients/heroku.js]: Can we not put this is every file?

File: src/clients/travis.js [hunk 2]
```diff
@@ -73,7 +74,7 @@ export async function getUserInformation(travisAccessToken, account) {
  * @returns {Promise}
  */
 export function getRepositoryId(travisAccessToken, username, projectName) {
-  winston.log('verbose', 'getRepositoryId', { username, projectName });
+  winston.log('verbose', 'getting repository id from travis', { username, projectName });
```
Comment (enclark) [inline src/clients/travis.js]: I don't know, I like the method name here better

Comment (jackcmeyer) [reply src/clients/travis.js]: I don't know, maybe we can do something like: `functionName : description`.

But at the end of the day, these messages would go to the user too. So a function name might not be useful.

File: src/clients/heroku.js (hunk 1 above)
Comment (jackcmeyer) [reply src/clients/heroku.js]: yeah, probably a good idea

File: src/clients/travis.js (hunk 2 above)
Comment (KrashLeviathan) [reply src/clients/travis.js]: I like the method name for debugging, but if we plan on giving the user a verbose command line option, it should be a phrase in standard English. So... maybe have two things that get logged: the method name at the `debug` level and the phrase at the `verbose` level?

File: src/clients/docker.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *  email: 'someemail@email.com',
+ *  password: 'somethingsomething'
+ * }
+ *
+ * @returns 'ok' if successful, otherwise returns the error
+ */
+// eslint-disable-next-line import/prefer-default-export
+export function login(dockerCredentials) {
+  const request = superagent
+    .get(`${dockerAuthApiUrl}`)
```
Comment (KrashLeviathan) [inline src/clients/docker.js]: Why use string interpolation here instead of just using the variable?

File: src/clients/github.js [hunk 4]
```diff
@@ -143,20 +124,20 @@ export function createGitHubRepository(projectName, projectDescription, token) {
  * @param projectName
  */
 export function initAddCommitAndPush(username, projectName, isTwoFactorAuth) {
-  winston.log('verbose', 'initAddCommitAndPush', { username, projectName });
-  winston.log('info', 'Pushing all files to the new git repository...');
+  winston.log('verbose', 'initialize github repo, create repo and push to repo', { username, projectName });
```
Comment (KrashLeviathan) [inline src/clients/github.js]: Keep the method name for this one, but change it to `debug` level. We are already giving the user information at the `info` level, and the verbose level seems a bit redundant.

File: src/clients/github.js [hunk 5]
```diff
@@ -143,20 +124,20 @@ export function createGitHubRepository(projectName, projectDescription, token) {
...
+  console.log(chalk.yellow('Pushing all files to the new git repository...'));
 
   return new Promise((resolve) => {
     if (!isTwoFactorAuth) {
       git(`${process.cwd()}/${projectName}`)
         .init()
         .add('.gitignore')
         .add('./*')
-        .commit('Initial commit')
+        .commit('Initial Commit w/ :heart: by @hammer-io.')
```
Comment (KrashLeviathan) [inline src/clients/github.js]: The changes here and below (line 150) reveal some duplicate code. Is it possible to move the `git.init()...commit()` outside the if-else block?

File: src/clients/github.js [hunk 6]
```diff
@@ -166,9 +147,11 @@ export function initAddCommitAndPush(username, projectName, isTwoFactorAuth) {
...
         .add('.gitignore')
         .add('./*')
-        .commit('Initial commit');
-      console.log('We cannot push hammer-io generated code to your repository because you have 2fa enabled. ' +
-        'Please follow this link (https://help.github.com/articles/providing-your-2fa-authentication-code/) for support.');
+        .commit('Initial Commit w/ :heart: by @hammer-io.');
+      console.log(chalk.red('We cannot push hammer-io generated code to your repository because' +
+      ' you have 2fa enabled. ' +
+        'Please follow this link' +
+        ' (https://help.github.com/articles/providing-your-2fa-authentication-code/) for support.'));
```
Comment (KrashLeviathan) [inline src/clients/github.js]: What are they supposed to do after visiting that link? Should we tell them to push manually with the git commands? Should they run tyr again?

File: src/utils/docker.js [hunk 7]
```diff
@@ -1,28 +1,49 @@
...
+ * @param credentials the docker credentials
+ *
+ * {
+ *  email: 'someemail@email.com',
+ *  password: 'somethingsomething"
+ * }
+ *
+ * @returns true if the login was successful, false if it was not
+ */
+export async function signInToDocker(credentials) {
```
Comment (KrashLeviathan) [inline src/utils/docker.js]: Recommend passing in username and password as separate parameters.

File: src/utils/heroku.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @returns user's heroku credentials if login was successful, false otherwise
+ *
+ * {
+ *  email: 'someemail@email.com',
+ *  password: 'somethingsomething'
+ * }
+ */
+// eslint-disable-next-line import/prefer-default-export
+export async function signInToHeroku(herokuCredentials) {
```
Comment (KrashLeviathan) [inline src/utils/heroku.js]: Recommend passing in email and password as separate parameters.

File: src/cli.js [hunk 9]
```diff
@@ -1,314 +1,222 @@
...
+
+    // sign in to third party tools
+    const credentials = await signInToThirdPartyTools(configs);
+    configs.credentials = credentials;
+
+    // initialize the basic project files
+    await initProject(configs);
+    console.log(chalk.green('!! Successfully generated your project!'));
+  } catch (err) {
+    console.log(chalk.red('!! Failed to gerneate your project!'));
```
Comment (KrashLeviathan) [inline src/cli.js]: gerneate

File: src/clients/github.js (hunk 5 above)
Comment (jackcmeyer) [reply src/clients/github.js]: @KrashLeviathan, I tried doing this once before. It did not work for me. I'll leave it as is for now, but we can open an issue for further research or just leave it.

Comment (KrashLeviathan) [reply src/clients/github.js]: :+1:

**Pull request merged**